crates/tauri-utils/src/resources.rs

```diff
@@ -132,122 +130,133 @@ pub struct ResourcePathsIter<'a> {
   /// whether the resource paths allows directories or not.
   allow_walk: bool,
 
-  /// The (key, value) of map when `pattern_iter` is a [`PatternIter::Map`],
-  /// used for determining [`Resource::target`]
-  current_pattern: Option<(String, PathBuf)>,
-
-  walk_iter: Option<walkdir::IntoIter>,
-  glob_iter: Option<glob::Paths>,
+  current_dest: Option<PathBuf>,
+  current_iter: Option<ResourcePathsInnerIter>,
 }
 
-impl ResourcePathsIter<'_> {
-  fn next_glob_iter(&mut self) -> Option<crate::Result<Resource>> {
-    let entry = self.glob_iter.as_mut().unwrap().next()?;
+#[derive(Debug)]
+enum ResourcePathsInnerIter {
+  Walk {
+    iter: walkdir::IntoIter,
+    current_pattern: Option<PathBuf>,
+  },
+  Glob {
+    iter: glob::Paths,
+  },
+}
 
-    let entry = match entry {
-      Ok(entry) => entry,
-      Err(err) => return Some(Err(err.into())),
-    };
+impl Iterator for ResourcePathsInnerIter {
+  type Item = crate::Result<PathBuf>;
 
-    self.next_current_path(normalize(&entry))
+  fn next(&mut self) -> Option<crate::Result<PathBuf>> {
+    match self {
+      ResourcePathsInnerIter::Walk { iter, .. } => Some(
+        iter
+          .next()?
+          .map(|entry| entry.into_path())
+          .map_err(Into::into),
+      ),
+      ResourcePathsInnerIter::Glob { iter } => Some(iter.next()?.map_err(Into::into)),
+    }
   }
+}
 
-  fn next_walk_iter(&mut self) -> Option<crate::Result<Resource>> {
-    let entry = self.walk_iter.as_mut().unwrap().next()?;
-
-    let entry = match entry {
-      Ok(entry) => entry,
-      Err(err) => return Some(Err(err.into())),
-    };
-
-    self.next_current_path(normalize(entry.path()))
+impl ResourcePathsIter<'_> {
+  fn next_current_iter(&mut self) -> Option<crate::Result<Resource>> {
+    let current_iter = self.current_iter.as_mut().unwrap();
+    let entry = current_iter.next()?;
+
+    Some(match entry {
+      Ok(entry) => {
+        // Skip directories
+        if entry.is_dir() {
+          self.next_current_iter()?
+        } else {
+          self.resource_from_path(normalize(&entry))
+        }
+      }
+      Err(error) => Err(error),
+    })
   }
 
-  fn resource_from_path(&mut self, path: &Path) -> crate::Result<Resource> {
+  fn resource_from_path(&self, path: PathBuf) -> crate::Result<Resource> {
     if !path.exists() {
-      return Err(crate::Error::ResourcePathNotFound(path.to_path_buf()));
+      return Err(crate::Error::ResourcePathNotFound(path));
     }
 
     Ok(Resource {
-      path: path.to_path_buf(),
-      target: if let Some((pattern, dest)) = &self.current_pattern {
-        // if processing a directory, preserve directory structure under current_dest
-        if self.walk_iter.is_some() {
-          dest.join(path.strip_prefix(pattern).unwrap_or(path))
-        } else if dest.components().count() == 0 {
```

Comment:
Is this possible? We have normalized this path and `join` for an empty `PathBuf` (`PathBuf::from("")`) should be the same as what's joined to anyways